src/main/kotlin/edu/carroll/polyominoes/service/game/GameServiceImpl.kt

```diff
@@ -0,0 +1,149 @@
+            return false
+        }
+
+        val leaderboard = Leaderboard(
+                stats.score, stats.level, stats.pieces, stats.rows, stats.time, LocalDateTime.now(
+                Clock.system(ZoneId.of("America/Denver"))
+        ), user
+        )
+
+        if (leaderboardRecord.isEmpty()) {
+            leaderboardRepo.save(leaderboard)
+            log.info("saveGameStats: Created game stats for {}", username)
+            return true
+        }
+
+        val existingLeaderboard = leaderboardRecord[0]
+        existingLeaderboard.score = stats.score
+        existingLeaderboard.level = stats.level
+        existingLeaderboard.pieces = stats.pieces
+        existingLeaderboard.rows = stats.rows
+        existingLeaderboard.time = stats.time
+        existingLeaderboard.date = LocalDateTime.now(Clock.system(ZoneId.of("America/Denver")))
+
+        leaderboardRepo.save(existingLeaderboard)
+
+        log.info("saveGameStats: Updated game stats for {}", username)
+
+        return true;
+    }
+
+    /**
+     *  When given a username and score checks if the score is the user's high score.
+     *
+     * @param username: A String representing the user to link the game statistics to.
+     * @param score: A Long representing the score a user got in the game.
+     *
+     * @return Returns true the score is a new high score for the user; false otherwise.
+     */
+    override fun checkHighScore(username: String, score: Long): Boolean {
+        val user = findUser(username);
+        if (user == null) {
+            log.debug("checkHighScore: found no users for '{}'", username)
+            return false
+        }
+
+        val scores = leaderboardRepo.findScoreByAccount(user);
+
+        if (scores.size > 1) {
+            log.warn("checkHighScore: found {} scores for '{}'", scores.size, username)
+            return false
+        }
+
+        if (scores.isEmpty()) {
+            log.info("checkHighScore: user '{}' had no prior score", username)
+            return true
+        }
+
+        val currentScore = scores[0].getScore()
+
+        return if (score > currentScore) {
+            log.info("checkHighScore: the score '{}' is a new high score for user '{}'", score, username)
+            true
+        } else if (score == currentScore) {
+            log.info("checkHighScore: the score '{}' is equal to the high score for user '{}'", score, username)
+            false
+        } else {
+            log.info("checkHighScore: the score '{}' is not a high score for user '{}'", score, username)
+            false
+        }
+
+    }
+
+    /**
+     * When given a username returns the corresponding Account
+     *
+     * @param  username: A String representing a user in the database.
+     *
+     * @return Returns an Account corresponding to the username; otherwise null;
+     */
+    private fun findUser(username: String): Account? {
+
+        if(username.isNullOrBlank()) {
```

Comment:
Style nit: Most of the time you have space after the `if` keyword, but not here.